File: Binary Search/153. Find Minimum in Rotated Sorted Array.c
int findMin(int* nums, int numsSize){
    int left = 0;
    int right = numsSize-1;
    int middle;
    int min = 5001;
    
    while (left<=right){
        middle = floor((left+right)/2);
        
        min = nums[middle] < min ? nums[middle] : min;
        
        if (nums[middle] < nums[right]){
            right = middle - 1;
        }
        else {
            left = middle + 1;
        }
    }
    
    return min;
}

File: Binary Search/153. Find Minimum in Rotated Sorted Array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left, right = 0, len(nums)-1 
        curr_min = float('inf')
        
        while left <= right:
            mid = (left+right)//2
            curr_min = min(curr_min, nums[mid])
            
            if nums[mid] < nums[right]: # min is on the left side
                right = mid - 1
            else: # min is on the right side 
                left = mid + 1
                
        return curr_min
